thunderbird-todoist/scripts/common.js

```diff
@@ -26,14 +34,51 @@ function fillAllProjectsSelect(selectid) {
             process(child, indent + 1);
           });
         }
-        const el = document.getElementById(selectid);
+        const el = document.getElementById(projectselect);
         el.innerHTML = "";
         projects.forEach((proj) => {
           process(proj, 0);
         });
       })
       .catch((err) => {
-        const el = document.getElementById(selectid);
+        const el = document.getElementById(projectselect);
+        el.innerHTML =
+          '<option value="0">Could not connect to Todoist...</option>';
+      });
+  });
+}
+
+function fillAllCollaboratorsSelect(collaboratorselect, projectid) {
+  loadDefaultCollaborator().then((selected) => {
+    getAllCollaborators(projectid)
+      .then((collaborators) => {
+        function process(collaborator, indent) {
+          let option = document.createElement("option");
+          let text = "";
+          for (let i = 0; i < indent; i++) {
+            text += "&nbsp;&nbsp;&nbsp;";
+          }
+          option.innerHTML = text + collaborator.name;
+          option.value = collaborator.id;
+          if (collaborator.id == selected) {
+            option.selected = true;
+          }
+          el.add(option);
+        }
```

Comment:
I think this can be simplified.

This function is needed for the projects because we can have nested projects, and we want child projects to have a bigger indentation than their parent. Therefore, we recursively call the `process` function with bigger indents.

In the case of collaborators we only have a single flat list of people. We don't need to be able to recursively descend or to indent some people; the value of `indent` is always `0`.

You can either remove the function alltogether and inline the required functionality, or just remove the unnecessary part from the function (i.e. the indentation part).